java-manta-client/src/main/java/com/joyent/manta/client/MantaObjectOutputStream.java

```diff
@@ -310,7 +310,10 @@ public synchronized void close() throws IOException {
         } catch (InterruptedException e) {
             // continue execution if interrupted
         } catch (ExecutionException e) {
-            throw new MantaIOException(e);
+            MantaIOException mioe = new MantaIOException(e);
```

Comment:
We can actually get it through the logging MDC threadlocal store. I'll update the PR with the request id added.